docs/examples/Grid/DataGrid.jsx

```diff
@@ -1,41 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
 import faker from 'faker';
 import { ComponentDisplay } from '../../components/ComponentDisplay';
 import { DataGrid, Column } from '../../../src';
 import { generateFakeDataArray } from '../../components/utils';
 import nanoid from 'nanoid';
+import { message } from 'antd';
 
 const generateFakeData = () => ({
     id: nanoid(10),
     name: faker.name.findName(),
-    age: faker.random.number({min: 10, max: 99}),
+    age: faker.random.number({ min: 10, max: 99 }),
     active: faker.random.boolean()
 });
 
-const data = generateFakeDataArray(5, generateFakeData);
+const defaultData = generateFakeDataArray(5, generateFakeData);
 
 // Example implementation
-const Example = (props) => (
-    <DataGrid
-        dataSource={data}
-        pagination={false}
-        onRecordCreate={() => {
-            return generateFakeData();
-        }}
-    >
-        <Column title={'Name'} dataIndex={'name'}/>
-        <Column title={'Age'} dataIndex={'age'} inputType={'number'}/>
-        <Column title={'Active'} dataIndex={'active'} inputType={'switch'}/>
-    </DataGrid>
-);
+const Example = () => {
+    const [data, setData] = useState(defaultData);
+
+    const onEdit = (id) => {
+        message.info(`edit button on row "${id}" clicked!`);
+    };
+
+    const onDelete = (ids) => {
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                setData(data.filter(rec => !ids.includes(rec.id)));
+                resolve();
+            }, 2000);
+        });
+    };
+
+    const onSave = (record) => {
+        const index = data.findIndex(rec => rec.id === record.id);
+
+        if (index === -1) {
+            data.push(record);
+        } else {
+            data[index] = record;
+        }
+
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                setData(data);
+                resolve();
+            }, 2000);
+        });
+    };
+
+    return (
+        <DataGrid
+            dataSource={data}
+            pagination={false}
+            onAdd={() => {
+                return generateFakeData();
+            }}
+            onEdit={onEdit}
+            onDelete={onDelete}
+            onSave={onSave}
+        >
+            <Column title={'Name'} dataIndex={'name'} fieldType={'string'}/>
+            <Column title={'Age'} dataIndex={'age'} fieldType={'number'}/>
+            <Column title={'Active'} dataIndex={'active'} fieldType={'boolean'}/>
```

Comment:
funktioniert die validierung auch? wenn ich jetzt z.b. ein required in einem column habe?

Reply:
ja, funktioniert. füge ich hinzu

Reply:
ok